src/core/qgsgdalutils.cpp

```diff
@@ -149,6 +150,43 @@ gdal::dataset_unique_ptr QgsGdalUtils::imageToMemoryDataset( const QImage &image
   return hSrcDS;
 }
 
+gdal::dataset_unique_ptr QgsGdalUtils::blockToMemoryDataset( int pixelWidth, int pixelHeight, const QgsRectangle &extent, void *block,  GDALDataType dataType )
```

Comment:
it would be good to make it clear in the function name that this creates a single-band dataset - e.g. `blockToSingleBandMemoryDataset(...)` ?